ui/apps/platform/src/hooks/useFetchNamespaceDeployments.ts

```diff
@@ -33,7 +44,78 @@ const DEPLOYMENTS_FOR_NAMESPACE_QUERY = gql`
     }
 `;
 
-function useFetchNamespaceDeployments(selectedNamespaceIds: string[]) {
+ */
+
+function useFetchNamespaceDeployments(
+    selectedNamespaceIds: string[],
+    namesToIdMapping: Map<string, string>
+) {
+    const [deploymentResponse, setDeploymentResponse] = useState<ListDeploymentResponse>({
+        loading: false,
+        error: '',
+        deploymentsByNamespace: [],
+    });
+    useEffect(() => {
+        if (selectedNamespaceIds.length > 0) {
+            setDeploymentResponse({
+                loading: true,
+                error: '',
+                deploymentsByNamespace: [],
+            });
+            const searchQuery: Record<string, string[]> = {
+                'Namespace ID': selectedNamespaceIds,
+            };
+            const sortOption = { field: 'Deployment', reversed: 'false' };
+            listDeployments(searchQuery, sortOption, 0, 0)
+                .then((response) => {
+                    const namespacesWithDeployments: NamespaceWithDeployments[] = [];
+                    const deploymentsByNamespace = new Map<string, Deployment[]>();
+                    response.forEach((listDeployment) => {
+                        const deployment: Deployment = {
+                            id: listDeployment.id,
+                            name: listDeployment.name,
+                        };
+                        const { namespace } = listDeployment;
```

Comment:
Here is an alternative that with object destructuring for all 3 variables in argument of `forEach` callback and slight rewriting:

```js
response.forEach({id, name, namespace}) => {
    const deployment = {id, name};
    if (deploymentsByNamespace.has(namespace)) {
        deploymentsByNamespace[namespace].push(deployment);
    } else {
        deploymentsByNamespace.set(namespace, [deployment]);
    }
});
```

Reply:
This piece was refactored a bit further. It seems the square bracket accessor breaks the flow.
I changed the conditions to always rely on `get` and `set` functions to interact with the map.